contrib/firebaseio/src/main/java/com/google/cloud/dataflow/contrib/firebase/io/FirebaseSource.java

```diff
@@ -366,14 +376,16 @@ protected Firebase getRef(){
 
   @Override
   public void validate() {
-
     AuthData result;
-
     try {
       result = auther.authenticate(this.getRef());
-    } catch (FirebaseException|InterruptedException e) {
+    } catch (FirebaseException e) {
+      result = null;
+      logger.error("Firebase authentication error", e);
```

Comment:
why not just throw in these cases, since you're going to throw below with a much less descriptive error message?